cmd/server-mux.go

```diff
@@ -215,18 +260,38 @@ func (m *ServerMux) ListenAndServeTLS(certFile, keyFile string) (err error) {
 
 	go m.handleServiceSignals()
 
-	listener, err := net.Listen("tcp", m.Server.Addr)
+	listeners, err := initListeners(m.Server.Addr, config)
 	if err != nil {
 		return err
 	}
 
-	listenerMux := &ListenerMux{Listener: listener, config: config}
-
 	m.mu.Lock()
-	m.listener = listenerMux
+	m.listeners = listeners
 	m.mu.Unlock()
 
-	err = http.Serve(listenerMux,
+	for _, listener := range listeners[:len(listeners)-1] {
+		go http.Serve(listener,
+			http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				// We reach here when ListenerMux.ConnMux is not wrapped with tls.Server
+				if r.TLS == nil {
+					u := url.URL{
+						Scheme:   "https",
+						Opaque:   r.URL.Opaque,
+						User:     r.URL.User,
+						Host:     r.Host,
+						Path:     r.URL.Path,
+						RawQuery: r.URL.RawQuery,
+						Fragment: r.URL.Fragment,
+					}
+					http.Redirect(w, r, u.String(), http.StatusMovedPermanently)
+				} else {
+					// Execute registered handlers
+					m.Server.Handler.ServeHTTP(w, r)
+				}
+			}),
+		)
+	}
+	err = http.Serve(listeners[len(listeners)-1],
```

Comment:
Check for error only on the last listener? Seems like we should check for all? It is worth commenting this.